add realroute parameter. route to amqp by default, or nowhere if "devnul...

fpeterschmitt opened the pull request:
...l"

Commit:
add realroute parameter. route to amqp by default, or nowhere if "devnull"

Commit:
processLine linked by default to processLineAMQP

Commit:
processLineDevNull

linkdd commented on tcp2canopsis/connector.py:
```diff
@@ -8,13 +8,22 @@
 
 import json
 
-
```
:camel: :warning: PEP08 wants two empty lines before classes

Commit:
pep8

Commit:
fix test_daemon_config_fail

Commit:
add unit tests for setRealRoute()

Commit:
fix unittests

As suggested by linkdd, get is better than set.

Add default route test and existence of processLine "attribute".

linkdd commented on tcp2canopsis/daemon.py:
```diff
@@ -29,9 +31,16 @@ def __init__(self, config, *args, **kwargs):
             raise RuntimeError('Invalid configuration: {0}'.format(err))
 
+        except RequiredFieldValidationError as err:
+            raise RuntimeError('Invalid configuration: {0}'.format(err))
```
:camel: :warning: Error message unclear, or merge the two catch with :

```python
except (FieldValidationError, RequiredFieldValidationError) as err:
```

linkdd commented on tcp2canopsis/connector.py:
```diff
@@ -53,11 +70,16 @@ def tryAuthentication(self, line):
             ))
 
-    def processLine(self, line):
+    processLine = None
```
:camel: :warning: What is the use of a class member ? You should put it in the constructor.

Commit:
missing required parameter message

Commit:
update readme with all configuration keys

Commit:
add cli realroute
